### PR TITLE
Support tasks in autocomplete

### LINKED ISSUE #526: Support Tasks in creation view

### LINKED ISSUE #527: Support tasks in edit view

### PR DESCRIPTION
**Pull request by willsb**
Closes #526

### CODE REVIEW

File: Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs [hunk 1]
```diff
@@ -183,9 +206,22 @@ public sealed class QueriesTheDatabaseForTimeEntries : AutocompleteProviderTest
...
                 [Fact]
+                public async Task SearchesTheTaskName()
+                {
+                    const string description = "21";
+                    var textFieldInfo = TextFieldInfo.Empty.WithTextAndCursor(description, 0);
+
+                    var suggestions = await Provider.Query(textFieldInfo);
+
+                    await Database.TimeEntries.Received().GetAll();
+                    suggestions.Should().HaveCount(1)
```
Comment (arturs105) [inline Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: I think we should check, whether the suggestion has a task with description "21".

Comment (arturs105) [reply Toggl.Foundation.Tests/Autocomplete/AutocompleteProviderTests.cs]: Upon closer inspection: `TimeEntrySuggestion` does not have any info about tasks, so disregard this.

**Comment by arturs105** [comment]
> according to Zeplin, the down arrow, which is shown next to the number of tasks of a project suggestion, should change to up arrow (or rotate 180 deg) when the list of tasks is expanded

This would require quite a lot of refactoring (making this PR even bigger), so i think it would be better if it was done in another PR.

> When I write the task name the time entry which has this task is show as suggestion. If I click on it, the project and the text is used, but if I save it, the new time entry does not have the task set, only the project

Yes, will fix this

**Comment by pe0ny** [comment]
> When I search by project ("@..."), I cannot search by the task name

Yes, it should search tasks as well.

> When I select some task, the token shows only the project although I would expect it to be "project: task". When I save it, it is OK though.

Yes. It should be 'Project: task'.

File: Toggl.Foundation.MvvmCross/ViewModels/StartTimeEntryViewModel.cs [hunk 2]
```diff
@@ -114,22 +117,31 @@ private void selectSuggestion(AutocompleteSuggestion suggestion)
...
+                        timeEntrySuggestion.ProjectId.Value,
+                        timeEntrySuggestion.ProjectName,
+                        timeEntrySuggestion.ProjectColor,
+                        timeEntrySuggestion.TaskId.Value,
+                        timeEntrySuggestion.TaskName);
                     break;
 
                 case ProjectSuggestion projectSuggestion:
-
+
```
Comment (simonrozsival) [inline Toggl.Foundation.MvvmCross/ViewModels/StartTimeEntryViewModel.cs]: These whitespaces shouldn't be here.

**Pull request merged**